mesheryctl/internal/cli/root/system/context.go

```diff
@@ -0,0 +1,399 @@
+	Channel       string   `mapstructure:"channel,omitempty"`
+	Version       string   `mapstructure:"version,omitempty"`
+}
+
+// createContextCmd represents the create command
+var createContextCmd = &cobra.Command{
+	Use:   "create <context-name>",
+	Short: "Create a new context (a named Meshery deployment)",
+	Long:  `Add a new context to Meshery config.yaml file`,
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		tempCntxt := utils.TemplateContext
+
+		if serverURL != "" {
+			err := utils.ValidateURL(serverURL)
+			if err != nil {
+				return err
+			}
+			tempCntxt.Endpoint = serverURL
+		}
+
+		log.Debug("serverURL: `" + tempCntxt.Endpoint + "`")
+
+		if platform != "" {
+			tempCntxt.Platform = platform
+		}
+
+		if len(adapters) >= 1 {
+			tempCntxt.Adapters = adapters
+		}
+
+		err := config.AddContextToConfig(args[0], tempCntxt, viper.ConfigFileUsed(), set)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("Added `%s` context", args[0])
+		return nil
+	},
+}
+
+// deleteContextCmd represents the delete command
+var deleteContextCmd = &cobra.Command{
+	Use:   "delete <context-name>",
+	Short: "delete context",
+	Long:  `Delete an existing context (a named Meshery deployment) from Meshery config file`,
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := viper.Unmarshal(&configuration)
+		if err != nil {
+			return err
+		}
+		_, exists := configuration.Contexts[args[0]]
+		if !exists {
+			return errors.New("no context to delete")
+		}
+
+		if viper.GetString("current-context") == args[0] {
+			var res bool
+			if utils.SilentFlag {
+				res = true
+			} else {
+				res = utils.AskForConfirmation("Are you sure you want to delete the current context")
+			}
+
+			if !res {
+				log.Printf("Delete aborted")
+				return nil
+			}
+
+			var result string
+
+			if newContext != "" {
+				_, exists := configuration.Contexts[newContext]
+				if !exists {
+					return errors.New("new context wrongly set")
+				}
+
+				if newContext == args[0] {
+					return errors.New("choose a new context other than the context being deleted")
+				}
+
+				result = newContext
+			} else {
+				var listContexts []string
+				for context := range configuration.Contexts {
+					if context != args[0] {
+						listContexts = append(listContexts, context)
+					}
+				}
+
+				prompt := promptui.Select{
+					Label: "Select context",
+					Items: listContexts,
+				}
+
+				_, result, err = prompt.Run()
+
+				if err != nil {
+					fmt.Printf("Prompt failed %v\n", err)
+					return err
+				}
+			}
+
+			fmt.Printf("The current context is now %q\n", result)
+			viper.Set("current-context", result)
+		}
+		delete(configuration.Contexts, args[0])
+		viper.Set("contexts", configuration.Contexts)
+		log.Printf("deleted context %s", args[0])
+		err = viper.WriteConfig()
+
+		return err
+	},
+}
+
+// listContextCmd represents the list command
+var listContextCmd = &cobra.Command{
```

Comment:
Could you add usage just like delete command?

Reply:
Sure 🙂

Reply:
Check the PR I am working on, or you can try hovering your mouse on `Use` word in vscode and it will show you how we can write `Use` to tell users how to use the command.